商品詳細表示機能実装

Yj-Satoshi opened the pull request:
What
商品詳細表示機能の実装
　(1)商品の各種情報の表示
　　（商品名などの基本情報に加え、出品者、カテゴリー、状態、配送に関する詳細内容の表示）

　(2)ログインの有無、ユーザー別の出品商品対する選択表示
　　・出品者本人（ログイン時）：「編集」・「削除」選択を表示
　　・出品者以外のユーザー（ログイン時）:「購入」選択のみ表示
　　・未ログイン時：商品情報の閲覧のみ（全ての選択非表示）

Why
商品の詳細の情報を表示させるページの実装の為

実装時の様子の写真は以下のGyazoのURLを参照願います。
 (1)出品者本人の場合
　　https://gyazo.com/91ebbeb453c49632fd105efd1e139f70
　　https://gyazo.com/a39e7e82bde625620dd72920ac766edb

 (2)出品者以外のユーザーの場合
　　https://gyazo.com/61f62cfed47b50b47c0b35841ded141e
　　https://gyazo.com/0ce9e02d93b7073a2acfb2ef8e688924

 (3)未ログイン時の場合
　　https://gyazo.com/483251a192e89f78d5f0ea8fe45ddf74
　　https://gyazo.com/f60c22698696167705c480df60ecb97b

Yj-Satoshi committed:
item詳細ページ遷移実装

Yj-Satoshi committed:
ダミー商品の削除

Yj-Satoshi committed:
詳細ページの表示の実装

Yj-Satoshi committed:
index.erb修正

商品写真の隙間がなくなった為、修正

Yj-Satoshi committed:
詳細ページの未ログイン時の表示修正

rubocop実行

i-date commented on app/views/items/index.html.erb:
```diff
@@ -129,8 +129,7 @@
       <li class='list'>
-
-        <%= link_to "#" do %>
+        <%= link_to  "/items/#{ item.id }" do %>
```
pathにprefix（`～_path`、`～`は適宜書き換え）を使用してはいかがでしょうか？
prefixを使用すると、任意のpathについて一意な命名になります。
それにより、pathがどこに遷移するのか分かりやすくなり、可読性や保守性の向上も見込めるようになります。

i-date commented on app/views/items/index.html.erb:
```diff
@@ -157,34 +156,11 @@
     <% end %> 
       <%# //商品のインスタンス変数になにか入っている場合、中身のすべてを展開できるようにしましょう %>
-
-      <%# 商品がない場合のダミー %>
-      <%# 商品がある場合は表示されないようにしましょう %>
-      <%# <li class='list'> %>
-
-        <%# <%= link_to '#' do %> 
-        <%# <%= image_tag "https://s3-ap-northeast-1.amazonaws.com/mercarimaster/uploads/captured_image/content/10/a004.png", class: "item-img" %> 
-        <%# <div class='item-info'>
-          <h3 class='item-name'>
-            商品を出品してね！
-          </h3>
-          <div class='item-price'>
-            <span>99999999円<br>(税込み)</span>
-            <div class='star-btn'> %>
-              <%# <%= image_tag "star.png", class:"star-icon" %> 
-              <%# <span class='star-count'>0</span>
-            </div>
-          </div>
-        </div> %>
-        <%# <% end %> 
-      <%# </li> %>
-      <%# //商品がある場合は表示されないようにしましょう %>
-      <%# /商品がない場合のダミー %>
-    <%# </ul> %>
```
ダミーコードを削除されていますが、こちらは実装案内で「商品がある場合は表示されないようにしましょう」とされていますが、商品が空の場合については表示されるようにしていただければと思います。
そのためif文を使用して、商品（`@items`）が空の場合には表示されるようにしましょう。

i-date commented on app/views/items/show.html.erb:
```diff
@@ -1,70 +1,73 @@
           <th class="detail-item">カテゴリー</th>
-          <td class="detail-value"><%= "カテゴリー名" %></td>
+          <td class="detail-value">
+            <%= CategoryChoice.data[(@item.category_id - 1)][:choice]  %> 
```
52行目ですが、アソシエーションを使用して`@item.category.name`のようにデータを取得できないでしょうか？
他のactive_hashを使用して管理しているデータについても、一度ご確認をお願いいたします。

i-date replied:
こちらのファイル内の108行目にも、下記のような「商品のカテゴリー名」を表示させるためのコードがありますので、こちらも「商品のカテゴリー名」が表示できるように修正しましょう。
```
<a href="#" class='another-item'><%= "商品のカテゴリー名" %>をもっと見る</a>
```

i-date reviewed (commented):
レビュー依頼ありがとうございます！
コメントいたしましたので、ご確認をお願いいたします。

Yj-Satoshi committed:
ActiveHashのViewのインスタンス取得の為の抜本修正

itemテーブルのActiveHashに関わるカラムのが本来〜_choice_idであるとこと、〜_idとしてた。
インスタンスでActiveHashの内容表示を取得すると、〜_choice_idの定義がないというNo Method Erorrが発生。
当エラー解消のため
・DBのリセットと各カラム名の修正
・モデルのテストコード、ビューファイルの影響する部分を修正

Yj-Satoshi committed:
pathのprefix修正とダミー商品表示

Yj-Satoshi committed:
詳細ファイル加筆修正

Yj-Satoshi commented:
i-date様
コメントありがとうございます。
ご指摘の箇所の修正ができましたので
確認の程、お願い致します。

なお今回の修正で商品詳細情報の表示に関わる部分を
itemテーブルから大きく修正しています。
表示の様子は以下Gyazoを参照願います。

https://gyazo.com/7c797cee1ba96c5a6fcad0c2aa7e1f51

ky3090 commented on app/views/items/show.html.erb:
```diff
@@ -1,70 +1,75 @@
           <th class="detail-item">カテゴリー</th>
-          <td class="detail-value"><%= "カテゴリー名" %></td>
+          <td class="detail-value">
+            <%# <%= CategoryChoice.data[(@item.category_id - 1)][:choice]  %>
```
52行目のコメントアウトが不要でしたら、削除しましょう！
理由としては、可読性向上のためです！

ky3090 reviewed (commented):
キャプチャ画像ならびに、レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

Yj-Satoshi committed:
コメントアウト削除

Yj-Satoshi commented:
yutakatahira様
レビューありがとうございます。
ご指摘いただいたコメントアウトを削除しましたので、
確認の程、お願い致します。

ky3090 commented on app/controllers/items_controller.rb:
```diff
@@ -1,13 +1,20 @@
+    
+    def show
+        @items = Item.order("created_at DESC")
+        # binding.pry
```
16行目のデバッグ用のコメントアウトも不要でしたら、削除しましょう！

ky3090 commented on spec/models/item_spec.rb:
```diff
@@ -6,7 +6,8 @@
   end
-  it "name, explanation, price, category_id, item_status_id, send_date_id, send_fee_id, imageがあれば出品登録できる" do
+  it "name, explanation, price, category_choice_id, item_status_choice_id, send_date_choice_id, send_fee_choice_id, imageがあれば出品登録できる" do
+    # binding.pry
```
10行目のでデバッグ用コードも削除しましょう！

ky3090 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

Yj-Satoshi committed:
コメントアウトの削除

Yj-Satoshi commented:
yutakatahira様
ご指摘ありがとうございます。
対象のコードを削除致しました。

Akinori0123 commented on app/controllers/items_controller.rb:
```diff
@@ -1,13 +1,19 @@
     end
+    
+    def show
+        @items = Item.order("created_at DESC")
```
```
@items = Item.order("created_at DESC")
```
【任意】
現在show.html.erbで@itemsが使用されていないため、今後使用する予定がなければ@itemsは消していただいても問題ないかと考えられます。
その結果showアクションの中身が空になる場合は「def show end」も消していただいて問題ありません。
こちら任意のタイミングでご確認お願いいたします。

Akinori0123 reviewed (commented):
レビュー依頼ありがとうございます。修正確認いたしました。
LGTMです。一点コメントいたしましたので任意のタイミングでご確認お願いいたします。
引き続き実装頑張っていきましょう！
